File: frontend/lib/core/models/learning_path.dart
class LearningPath {
  final int id;
  final String code;
  final String nameAr;
  final String nameFr;
  final String description;

  const LearningPath({
    required this.id,
    required this.code,
    required this.nameAr,
    required this.nameFr,
    required this.description,
  });

  factory LearningPath.fromJson(Map<String, dynamic> json) {
    return LearningPath(
      id: json['id'] as int,
      code: json['code'] as String,
      nameAr: json['name_ar'] as String,
      nameFr: json['name_fr'] as String,
      description: json['description'] as String? ?? '',
    );
  }
}

class BacBranch {
  final int id;
  final String code;
  final String nameAr;
  final String nameFr;

  const BacBranch({
    required this.id,
    required this.code,
    required this.nameAr,
    required this.nameFr,
  });

  factory BacBranch.fromJson(Map<String, dynamic> json) {
    return BacBranch(
      id: json['id'] as int,
      code: json['code'] as String,
      nameAr: json['name_ar'] as String,
      nameFr: json['name_fr'] as String,
    );
  }
}

File: backend/internal/models/learning_path.go
package models

import "time"

type LearningPath struct {
	ID          uint      `json:"id"`
	Code        string    `json:"code"`
	NameAr      string    `json:"name_ar"`
	NameFr      string    `json:"name_fr"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
}

// AdminLearningPath includes the enabled flag, only exposed to the admin panel.
type AdminLearningPath struct {
	ID          uint      `json:"id"`
	Code        string    `json:"code"`
	NameAr      string    `json:"name_ar"`
	NameFr      string    `json:"name_fr"`
	Description string    `json:"description"`
	Enabled     bool      `json:"enabled"`
	CreatedAt   time.Time `json:"created_at"`
}

type BacBranch struct {
	ID     uint   `json:"id"`
	Code   string `json:"code"`
	NameAr string `json:"name_ar"`
	NameFr string `json:"name_fr"`
}
